### PR TITLE
Replat ComponentHubReliabilityTests on SignalR test infra

### PR DESCRIPTION
**Pull request by captainsafia**
Addresses https://github.com/dotnet/aspnetcore/issues/19666.

Trying to use the same testing server that SignalR uses for the hub connection tests to see if it will improve stability for the component hub tests. We're switching to using a single in-process server per test scenario.

### CODE REVIEW

File: src/Components/.vscode/launch.json [hunk 1]
```diff
@@ -1,6 +1,12 @@
 {
   "version": "0.2.0",
   "configurations": [
+    {
+      "name": ".NET Core Attach",
+      "type": "coreclr",
+      "request": "attach",
+      "processId": "${command:pickProcess}"
+    },
```
Comment (TanayParikh) [inline src/Components/.vscode/launch.json:9]: Was this intentional?

File: src/Shared/SignalR/InProcessTestServer.cs [hunk 2]
```diff
@@ -15,6 +15,7 @@
 using Microsoft.Extensions.DependencyInjection;
 using Microsoft.Extensions.Hosting;
 using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Logging.Console;
```
Comment (BrennanConroy) [inline src/Shared/SignalR/InProcessTestServer.cs:18]: remove?

**Comment by HaoK** [comment]
Are the component tests currently sharing servers processes for our e2e tests?  Is this change for the entire project or only these signal r reliability tests? (seems good to switch to single server per test generally anyways)

File: src/Components/test/E2ETest/ServerExecutionTests/ComponentHubReliabilityTest.cs [hunk 3]
```diff
@@ -247,55 +286,72 @@ public async Task OnLocationChanged_ReportsDebugForExceptionInValidation()
...
-            var entry = Assert.Single(Logs, l => l.EventId.Name == "LocationChangeFailed");
-            Assert.Equal(LogLevel.Error, entry.LogLevel);
-            Assert.Matches($"Location change to '{new Uri(ServerFixture.RootUri, "/test")}' in circuit '.*' failed\\.", entry.Message);
+            await using (var server = await StartServer<ServerStartup>(verifyLogs: false))
+            {
+                // while (!System.Diagnostics.Debugger.IsAttached)
+                // {
+                //     System.Console.WriteLine($"Waiting to attach on ${System.Diagnostics.Process.GetCurrentProcess().Id}");
+                //     System.Threading.Thread.Sleep(1000);
+                // }
```
Comment (TanayParikh) [inline src/Components/test/E2ETest/ServerExecutionTests/ComponentHubReliabilityTest.cs:330]: Remove?

**Review by TanayParikh — APPROVED** [review]
LGTM

File: src/Shared/SignalR/FunctionalTestBase.cs [hunk 4]
```diff
@@ -36,9 +36,9 @@ public FunctionalTestBase()
             };
         }
 
-        public Task<InProcessTestServer<T>> StartServer<T>(Func<WriteContext, bool> expectedErrorsFilter = null) where T : class
+        public Task<InProcessTestServer<T>> StartServer<T>(Func<WriteContext, bool> expectedErrorsFilter = null, bool verifyLogs = true) where T : class
```
Comment (BrennanConroy) [inline src/Shared/SignalR/FunctionalTestBase.cs:39]: Can the tests that use this instead pass in a func that just returns true?

File: src/Components/test/E2ETest/ServerExecutionTests/ComponentHubReliabilityTest.cs (hunk 3 above)
Comment (captainsafia) [reply src/Components/test/E2ETest/ServerExecutionTests/ComponentHubReliabilityTest.cs:330]: *faceplam* Good catch.

File: src/Components/.vscode/launch.json (hunk 1 above)
Comment (captainsafia) [reply src/Components/.vscode/launch.json:9]: Yeah, this is helpful if you want to attach to debug tests in the src/Components area.

File: src/Shared/SignalR/FunctionalTestBase.cs (hunk 4 above)
Comment (captainsafia) [reply src/Shared/SignalR/FunctionalTestBase.cs:39]: Ah -- yep! That would work.

**Comment by captainsafia** [comment]
> Are the component tests currently sharing servers processes for our e2e tests? 

I think it's a test server per test class running in a background thread. Mostly, I'm trying to rule out if any of the Blazor-host specific stuff we do might be causing the issue.

>  Is this change for the entire project or only these signal r reliability tests?

This change is only for the test class with the most number of flaky tests. If it shows promise, I'll expand it to the other tests in the server reliability area.

**Review by BrennanConroy — APPROVED** [review]
Hopefully this will make things better!

**Pull request merged**